spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -258,6 +258,10 @@ object DistriOptimizer {
             val taskSize = gradLength / _subModelNumber
             val extraTask = gradLength % _subModelNumber
 
+            (0 until _subModelNumber).diff(finishedThreads).foreach(i =>
```

Comment:
Please give more description of the fix. I don't know why make this change without more background.